crmd/lrm_state.c

```diff
@@ -222,6 +229,11 @@ lrm_state_reset_tables(lrm_state_t * lrm_state)
                   g_hash_table_size(lrm_state->rsc_info_cache));
         g_hash_table_remove_all(lrm_state->rsc_info_cache);
     }
+    if (lrm_state->metadata_cache) {
+        crm_trace("Re-setting metadata cache with %d members",
+                  g_hash_table_size(lrm_state->metadata_cache));
+        g_hash_table_remove_all(lrm_state->metadata_cache);
+    }
```

Comment:
Basically -- but since this is not a public API function, we can simply add a TRUE/FALSE argument to specify whether meta-data should be cleared.

I'm thinking crmd/lrm.c:do_lrm_control() doesn't need to clear it either. It's not called for remote nodes, so that's not an issue, but I don't think the meta-data becomes invalid just because we disconnect from the local lrmd. I don't know whether it's possible to record operations after getting to that point, but it's worth keeping the meta-data just in case.